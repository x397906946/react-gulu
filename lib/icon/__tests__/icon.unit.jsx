import React from 'react';
import Icon from '../icon';
import * as renderer from 'react-test-renderer';
import {mount} from 'enzyme';

describe('Icon', () => {
  it('render success', () => {
    const json = renderer.create(<Icon/>).toJSON();
    expect(json).toMatchSnapshot();
  });
  it('onClick', () => {
    const fn = jest.fn();
    const component = mount(<Icon name="alipay" onClick={fn}/>);
    component.find('svg').simulate('click');
    expect(fn).toBeCalled();
  });
});
