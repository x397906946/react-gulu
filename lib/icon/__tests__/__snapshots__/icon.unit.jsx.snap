// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon render success 1`] = `
<svg
  className="x-icon"
>
  <use
    xlinkHref="#undefined"
  />
</svg>
`;
